.github/workflows/nightly-build.yml

```diff
@@ -5,21 +5,58 @@ on:
     inputs:
       release-branch:
         description: 'Release branch to bump and merge (leave empty to skip)'
+        required: false
+        type: string
+      token:
+        description: 'Token with push permissions'
+        required: true
         type: string
-  schedule:
-    - cron: '0 4 * * *'
 
 jobs:
+  bump-and-merge:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Configure Git
+        run: |
+          git config --local user.name "Nightly Build Action"
+          git config --local user.email "action@github.com"
+          git remote set-url origin https://x-access-token:${{ inputs.token }}@github.com/${{ github.repository }}.git
+
+      - name: Merge release branch into main
+        if: inputs.release-branch != ''
+        continue-on-error: true  # Always version bump even if merge fails
+        run: |
+          git checkout main
+          git merge origin/${{ inputs.release-branch }}
+          git push origin main
```

Comment:
If this fails, what should we do? The best I could come up with was having the workflow create a PR? Does that make sense or is there a better plan?